src/rectangle.js

```diff
@@ -223,67 +248,79 @@ $.Rect.prototype = /** @lends OpenSeadragon.Rect.prototype */{
      */
     // ----------
     union: function(rect) {
+        if (this.degrees !== 0 || rect.degrees !== 0) {
```

Comment:
Let go for that second approach.
Maybe always returning the union of the bounding boxes would be more consistent though?